Core/Business/Dispatcher/BaseEventDispatcher.php

```diff
@@ -0,0 +1,128 @@
+ * except the Event instance is a HookEvent, and contains hook parameters,
+ * and can keep the listener response as attribute.
+ *
+ * Existing dispatchers:
+ * The default ones: @see \PrestaShop\PrestaShop\Core\Foundation\Dispatcher\EventDispatcher
+ * New ones are defined:
+ *
+ * - module         All events concerning modules manipulation: install, update, uninstall, etc...
+ *                  FIXME: FOR NOW, THESE EVENTS ARE NOT TRIGGERED. FOR FUTURE BEHAVIOR.
+ *      - before_install
+ *      - after_install
+ *      - before_update
+ *      - after_update
+ *      - before_uninstall
+ *      - after_uninstall
+ *      - before_deactivate
+ *      - after_deactivate
+ *      - before_reactivate
+ *      - after_reactivate
+ */
+final class BaseEventDispatcher extends EventDispatcher
+{
+    /**
+     * PrestaShop Business specific listeners.
+     *
+     * This static attribute contains PrestaShop Business specific listeners to register during Router instantiation.
+     * The array will be merged into the super class array to complete the list.
+     * @var array
+     */
+    private static $baseDispatcherRegistry = array(
+        'module' => array(
+            array('before_install', 'PrestaShop\\PrestaShop\\Core\\Business\\Module\\ModuleEventListener', 'onBefore', -255, false),
+            array('before_update', 'PrestaShop\\PrestaShop\\Core\\Business\\Module\\ModuleEventListener', 'onBefore', -255, false),
+            array('before_uninstall', 'PrestaShop\\PrestaShop\\Core\\Business\\Module\\ModuleEventListener', 'onBefore', -255, false),
+            array('before_deactivate', 'PrestaShop\\PrestaShop\\Core\\Business\\Module\\ModuleEventListener', 'onBefore', -255, false),
+            array('before_reactivate', 'PrestaShop\\PrestaShop\\Core\\Business\\Module\\ModuleEventListener', 'onBefore', -255, false),
+            array('after_install', 'PrestaShop\\PrestaShop\\Core\\Business\\Module\\ModuleEventListener', 'onAfter', 128, false),
+            array('after_update', 'PrestaShop\\PrestaShop\\Core\\Business\\Module\\ModuleEventListener', 'onAfter', 128, false),
+            array('after_uninstall', 'PrestaShop\\PrestaShop\\Core\\Business\\Module\\ModuleEventListener', 'onAfter', 128, false),
+            array('after_deactivate', 'PrestaShop\\PrestaShop\\Core\\Business\\Module\\ModuleEventListener', 'onAfter', 128, false),
+            array('after_reactivate', 'PrestaShop\\PrestaShop\\Core\\Business\\Module\\ModuleEventListener', 'onAfter', 128, false),
+        ), // all events concerning modules manipulation: install, update, uninstall, etc...
+        'hook' => array(
+            array('legacy_actionProductAdd', 'Adapter_HookManager', 'onHook', 0, true),
+            array('legacy_actionProductUpdate', 'Adapter_HookManager', 'onHook', 0, true),
+            array('legacy_actionCategoryUpdate', 'Adapter_HookManager', 'onHook', 0, true),
+            // TODO: complete this list for Admin Product page
+        ) // hooks
```

Comment:
Just to keep a note of it: Maybe a external config php array with an include on class ctor.